devito/types/dense.py

```diff
@@ -998,7 +998,15 @@ def __init_finalize__(self, *args, **kwargs):
         else:
             raise TypeError("`space_order` must be int or 3-tuple of ints")
 
-        self._fd = self.__fd_setup__()
+        # Acquire derivative shortcuts
+        if self is self.function:
+            self._fd = self.__fd_setup__()
```

Comment:
Nice forgot about that yeah. 

Would check wasn't (unsafely) rebuilt with a different space order maybe?